### PR TITLE
Add option `distributor.ingestion-burst-factor`

### PR DESCRIPTION
**Pull request by treid314**
#### What this PR does

This PR is to add `distributor.ingestion-burst-factor`, when set it will override the `distributor.ingestion-burst-size`. When `distributor.ingestion-burst-factor` is set the burst size on the underlying token bucket for the ingest rate limiter to a multiple of the maximum ingestion rate a distributor can handle. This is to simplify setting the burst size to a multiple of the global rate limit. 

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [X] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/distributor/rate_strategy.go
Comment (treid314) [reply pkg/distributor/rate_strategy.go:22]: We use globalStrategy for [request rate limit strategy](https://github.com/grafana/mimir/blob/main/pkg/distributor/distributor.go#L424) as well which doesn't use the burst factor so we can't combine `globalStrategy` with `ingestionRateStrategy`,  but we can possibly remove the IngestionRateStrategy.

File: pkg/util/validation/limits.go
Comment (treid314) [reply pkg/util/validation/limits.go:197]: I think it's simpler to keep this as a whole number, you'd always want this to be at least 1 so that your burst is larger than your ingest limit. You can make the case for having a range between 1 and 2 but I don't really see us tuning this number very much unlike the rate limit.

File: pkg/distributor/distributor.go
Comment (pracucci) [inline pkg/distributor/distributor.go:959]: Have you considered just moving this logic to `IngestionBurstSize()`? Basically the `IngestionBurstSize()` can just return the burst size to apply, which could be the one configured or the new one based on the factor.

**Review by pracucci — COMMENTED** [review]
Thanks for working on this. I'm happy to see the burst factor config option, cause it set exactly what the burst is.

File: pkg/distributor/distributor.go
Comment (treid314) [reply pkg/distributor/distributor.go:959]: I'm not sure if we can do that because we divide the burst factor result by the number of distributors and we do not do that with the burst size in the rate limit strategy.

File: pkg/util/validation/limits.go
Comment (treid314) [inline pkg/util/validation/limits.go:451]: I'm not a big fan of silently setting this to 0 if the input is less than 1. Based on the other configs this doesn't seem too far off what we do in other places but I'm not sure if there's a better way.

Comment (pracucci) [reply pkg/util/validation/limits.go:451]: I don't feel strong about this, but I think what you're doing is fine. I don't see many alternatives.

File: pkg/distributor/distributor_test.go
Comment (pracucci) [inline pkg/distributor/distributor_test.go:626]: [nit] Could you describe the scenario you're testing, please?

File: pkg/distributor/rate_strategy_test.go
Comment (pracucci) [inline pkg/distributor/rate_strategy_test.go:61]: We have a real use case with `rate.Inf`. I would test it too.

**Review by pracucci — APPROVED** [review]
Good job. The change LGTM. I left a couple of nits. No need to wait for another review from myself, unless the code will significantly change.

File: pkg/distributor/rate_strategy.go
Comment (flxbk) [inline pkg/distributor/rate_strategy.go:57]: Why are we not dividing the burst size up across distributors, like we are doing with the limit?

Comment (flxbk) [reply pkg/distributor/rate_strategy.go:57]: Never mind, I misread the diff :)

**Pull request merged**
